pipeline-templates/job/build-and-train.yml

```diff
@@ -0,0 +1,50 @@
+parameters:
+  Environment:
+  ServiceConnection:
+
+jobs:
+  - deployment: DeployInfra
+    environment: ${{ parameters.Environment }}
+    strategy:
+      runOnce:
+        deploy:
+          steps:
+            - checkout: self
+            - template: azure-pipelines-templates/deploy/step/arm-deploy.yml@das-platform-building-blocks
+              parameters:
+                ServiceConnection: ${{ parameters.ServiceConnection }}
+                SubscriptionId: $(SubscriptionId)
+                Location: $(ResourceGroupLocation)
+                Environment: ${{ parameters.Environment }}
+                TemplatePath: das-data-management-ai/azure/template.json
+                ParametersPath: das-data-management-ai/azure/template.parameters.json
+                IsMultiRepoCheckout: true
+  - job: BuildAndTrain
+    dependsOn: DeployInfra
+    variables:
+      - name: WorkspaceName
+        value: $[ dependencies.DeployInfra.outputs['DeployInfra.ArmOutputs${{ parameters.Environment }}.WorkspaceName'] ]
+      - name: ResourceGroupName
+        value: $[ dependencies.DeployInfra.outputs['DeployInfra.ArmOutputs${{ parameters.Environment }}.ResourceGroupName'] ]
+      - name: ComputeName
+        value: $[ dependencies.DeployInfra.outputs['DeployInfra.ArmOutputs${{ parameters.Environment }}.ComputeName'] ]
+    steps:
+      - checkout: self
+      - task: AzureCLI@2
+        displayName: "Train and Register"
+        inputs:

```

Comment:
This should be refactored but on the assumption we'll be returning to this pipeline that can (and probably should) wait till it's been used a bit more by the team

Reply:
Yes agreed, I don't think this is the long-term solution but needs to get some use to see where the issues are